maro/simulator/scenarios/vm_scheduling/virtual_machine.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List
+
+
+class VirtualMachine:
+    """VM object.
+
+    The VM lifetime equals to the deletion tick - creation tick.
+    For example:
+        A VM's cpu utilization is {tick: cpu_utilization} = {0: 0.1, 1: 0.4, 2: 0.2}.
+        In our scenario, we define vm's creation tick as 0, and deletion tick as 3.
+        Its lifetime will be deletion tick - creation tick = 3 - 0 = 3
+
+    Args:
+        id (int): The VM id.
+        cpu_cores_requirement (int): The amount of virtual cores requested by VM.
+        memory_requirement (int): The memory requested by VM. The unit is (GBs).
+        lifetime (int): The lifetime of the VM, that is, deletion tick - creation tick + 1.
+    """
+    def __init__(self, id: int, cpu_cores_requirement: int, memory_requirement: int, lifetime: int):
+        # VM Requirement parameters.
+        self.id: int = id
+        self.cpu_cores_requirement: int = cpu_cores_requirement
+        self.memory_requirement: int = memory_requirement
+        self.lifetime: int = lifetime
+        # VM utilization list with VM cpu utilization(%) in corresponding tick.
+        self._utilization_series: List[float] = []
+        # The physical machine Id that the VM is assigned.
+        self.pm_id: int = -1
+        self._cpu_utilization: float = 0.0
+        self.creation_tick: int = -1
+        self.deletion_tick: int = -1
+
+    @property
+    def cpu_utilization(self) -> float:
+        return self._cpu_utilization
+
+    @cpu_utilization.setter
+    def cpu_utilization(self, cpu_utilization: float):
+        self._cpu_utilization = min(max(0, cpu_utilization), 100)
+
+    def get_utilization(self, cur_tick: int):
```

Comment:
what's the difference between the get_utilization and cpu_utilization?
or rename to get_cpu_utilization_by_tick, and you can support any tick in the [c_t, d_t]

Reply:
So get_utilization is aim to get the cpu utilization from the utilization series. 
A VM object contains a `cpu_utilization` and a `utilization_series`. 
After reading the cpu values from the data file, we store the utilization in the utilization series. And we can only query the utilization from the utilization series through `get_utilization`. And then we save the value into `cpu_utilization`.